### PR TITLE
Fix PowerRegenEvent not being cancellable + add delta to event

### PR DESCRIPTION
**Pull request by MedicOP**
See https://github.com/SavageLLC/SavageFactions/issues/230

### CODE REVIEW

**Comment by prosavage** [comment]
Nice contribution I'm sure It'll help others

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java [hunk 1]
```diff
@@ -599,14 +599,18 @@ public void updatePower() {
             return;  // don't let dead players regain power until they respawn
         }
 
-        PowerRegenEvent powerRegenEvent = new PowerRegenEvent(getFaction(), this);
+        double delta = millisPassed * Conf.powerPerMinute / 60000; // millisPerMinute : 60 * 1000
```
Comment (liz3) [inline src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java:602]: Is `Conf.powerPerMinute` null safe?

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java [hunk 2]
```diff
@@ -599,14 +599,18 @@ public void updatePower() {
...
         if (Bukkit.getPluginManager().getPlugin("SavageFactions") != null) {
-            Bukkit.getScheduler().runTask(SavageFactions.plugin, () -> Bukkit.getServer().getPluginManager().callEvent(powerRegenEvent));
-
+            Bukkit.getScheduler().runTask(SavageFactions.plugin, () -> {
+                PowerRegenEvent powerRegenEvent = new PowerRegenEvent(getFaction(), this, delta);
```
Comment (liz3) [inline src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java:605]: same for `getFaction()`

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java (hunk 1 above)
Comment (MedicOP) [reply src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java:602]: it's a primitive..

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java (hunk 2 above)
Comment (MedicOP) [reply src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java:605]: that code was already there

**Comment by MedicOP** [comment]
> 
> 
> Nice contribution I'm sure It'll help others

Will it be merged?

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java (hunk 1 above)
Comment (liz3) [reply src/main/java/com/massivecraft/factions/zcore/persist/MemoryFPlayer.java:602]: thats why i asked :P

**Pull request merged**